### PR TITLE
Update 107-string.md

### PR DESCRIPTION
**Pull request by hiahir357**
I've added some useful links to the string step of the Rust roadmap

### CODE REVIEW

File: src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md [hunk 1]
```diff
@@ -4,4 +4,9 @@ In Rust, `String` is a growable, mutable, owned, UTF-8 encoded string type. When
 
 Learn more from the following links:
 
-- [@article@What Is a String?](https://doc.rust-lang.org/book/ch08-02-strings.html?highlight=String#what-is-a-string)
\ No newline at end of file
+- [@official@String in std::string - Rust](https://doc.rust-lang.org/std/string/struct.String.html)
+- [@official@str - Rust](https://doc.rust-lang.org/std/primitive.str.html)
+- [@article@What Is a String?](https://doc.rust-lang.org/book/ch08-02-strings.html?highlight=String#what-is-a-string)
+- [@article@Rust String (With Examples)](https://www.programiz.com/rust/string)
+- [@article@Differences between String and str in Rust](https://stackoverflow.com/questions/24158114/what-are-the-differences-between-rusts-string-and-str)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md:11]: Lets remove the SO link

**Review by dansholds — CHANGES_REQUESTED** [review]

File: src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md (hunk 1 above)
Comment (dansholds) [inline src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md:11]:
```suggestion
```

File: src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md [hunk 2]
```diff
@@ -4,4 +4,8 @@ In Rust, `String` is a growable, mutable, owned, UTF-8 encoded string type. When
 
 Learn more from the following links:
 
-- [@article@What Is a String?](https://doc.rust-lang.org/book/ch08-02-strings.html?highlight=String#what-is-a-string)
\ No newline at end of file
+- [@official@String in std::string - Rust](https://doc.rust-lang.org/std/string/struct.String.html)
+- [@official@str - Rust](https://doc.rust-lang.org/std/primitive.str.html)
+- [@article@What Is a String?](https://doc.rust-lang.org/book/ch08-02-strings.html?highlight=String#what-is-a-string)
```
Comment (dansholds) [inline src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md:9]:
```suggestion
- [@official@What Is a String?](https://doc.rust-lang.org/book/ch08-02-strings.html?highlight=String#what-is-a-string)
```

File: src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md (hunk 1 above)
Comment (hiahir357) [reply src/data/roadmaps/rust/content/101-language-basics/103-data-structures/107-string.md]: :+1:

**Pull request merged**